huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go

```diff
@@ -0,0 +1,250 @@
+package modelarts
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceModels_basic(t *testing.T) {
+	rName := "data.huaweicloud_modelarts_models.name_filter"
+	dc := acceptance.InitDataSourceCheck(rName)
+	name := acceptance.RandomAccResourceName()
+	name2 := acceptance.RandomAccResourceName()
+	bucketName := acceptance.RandomAccResourceNameWithDash()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceModels_basic(name, name2, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrPair(rName, "models.0.id", "huaweicloud_modelarts_model.test1", "id"),
+					resource.TestCheckResourceAttr(rName, "models.0.name", name),
+					resource.TestCheckResourceAttr(rName, "models.0.version", "0.0.1"),
+					resource.TestCheckResourceAttr(rName, "models.0.model_type", "TensorFlow"),
+					resource.TestCheckResourceAttr(rName, "models.0.description", "This is a demo"),
+					resource.TestCheckResourceAttr(rName, "models.0.workspace_id", "0"),
+					resource.TestCheckResourceAttr(rName, "models.0.status", "published"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.model_source"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.install_type.#"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.size"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.market_flag"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.tunable"),
+					resource.TestCheckResourceAttrSet(rName, "models.0.publishable_flag"),
+
+					resource.TestCheckOutput("default_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("name_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("version_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("status_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("description_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("workspace_id_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("model_type_filter_is_useful", "true"),
+
+					resource.TestCheckOutput("not_model_type_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceModels_basic(name, name2, bucketName string) string {
+	modelConfig := testAccDatasourceModels_config(name, name2, bucketName)
+
+	return fmt.Sprintf(`
+%[1]s
+
+data "huaweicloud_modelarts_models" "test" {
+}
+
+output "default_filter_is_useful" {
+  value = length(data.huaweicloud_modelarts_models.test.models) > 2
+}
+
+data "huaweicloud_modelarts_models" "name_filter" {
+  name        = huaweicloud_modelarts_model.test1.name
+  exact_match = "true"
+}
```

Comment:
missing `depends_on` definition, this data source will execute ahead of the resource.
and the object of the alltrue function will be an empty slice, even if the result shows true

Reply:
ok, add depends_on